### PR TITLE
fixes #5436 - provisioning templates are world accessible

### PR DESCRIPTION
**Pull request by ohadlevy**

### CODE REVIEW

File: app/controllers/unattended_controller.rb [hunk 1]
```diff
@@ -103,8 +103,7 @@ def get_host_details
   end
 
   def find_host_by_spoof
-    host   = Host.find_by_ip(params.delete('spoof')) if params['spoof'].present?
-    host ||= Host.find_by_name(params.delete('hostname')) if params['hostname'].present?
+    host   = Host.where(['ip = ? OR name = ?', params[:spoof], params[:hostname]]).first
```
Comment (domcleal) [inline app/controllers/unattended_controller.rb]: One bug we fixed here a while ago was when hosts existed without IP addresses, we'd be searching where `ip IS NULL`.  It looks like this change might reintroduce the bug, as we could generate a search for:

```
WHERE ip IS NULL OR name = 'foo.example.com'
```

Which could return the first (random) host without an IP address.

Comment (domcleal) [inline app/controllers/unattended_controller.rb]: see 1e0fd28

**Comment by ohadlevy** [comment]
updated.

**Comment by domcleal** [comment]
Thanks @ohadlevy, merged as aa0ebe8 for Foreman 1.5.0.

**Pull request closed without merging**